### PR TITLE
Redo the json-ld logic

### PR DESCRIPTION
**Pull request by dbrower**
The old algorithm looked at each triple one at a time, and dealt with
the blank nodes as they appeared. This code uses the json/ld gem to
group predicates by subject and then shortens URIs based on our own
prefix table. And triples whose object is a blank node are dealt with
when encountered. Once each triple with the target subject has been
looked at, we map them into the accumulator and use that to generate the
ROF file.

This simplifies the code and makes the parsing and ROF generation two
clearly separate steps.

It also relaxes the error checking and URIs that have prefixes not in
our NAMESPACE map are allowed as predicates (instead of raising an error).
Unknown predicates all end up in the metadata section of the ROF. This
lets us add more predicates as needed without needing to update the code.

### CODE REVIEW

File: lib/rof/translators/jsonld_to_rof.rb [hunk 1]
```diff
@@ -16,71 +15,20 @@ module Translators
...
+      # since RDF nominally uses full URIs, but we prefer the shorter labels
+      # using the tags.
+      NAMESPACES = {
+        'http://purl.org/ontology/bibo/' => 'bibo:',
+        'info:fedora/fedora-system:def/relations-external#' => 'rels-ext:',
+        'http://id.loc.gov/vocabulary/relators/' => 'mrel:',
+        'http://www.ebu.ch/metadata/ontologies/ebucore/ebucore#' => 'ebucore:',
+        'https://library.nd.edu/ns/terms/' => 'nd:',
+        'http://purl.org/dc/terms/' => 'dc:',
+        'http://projecthydra.org/ns/relations#' => 'hydra-rel:',
```
Comment (laritakr) [inline lib/rof/translators/jsonld_to_rof.rb:28]: Why was this changed to hydra-rel from hydramata-rel?

**Review by laritakr — APPROVED** [review]
I don't totally follow everything, but most of it makes sense and seems much more straightforward than before. I'm always for code changes to make things clearer!

**Comment by dbrower** [comment]
@laritakr your deleted comment suggested an improvement I could make. I’m going to do that this afternoon.

**Comment by laritakr** [comment]
Ok! I didn't realize it would trigger an improvement... I just wasn't initially following the change to hydra-rel from hydramata-rel.

**Comment by dbrower** [comment]
ok, I renamed the `hydra-rels` prefix, which was just a placeholder, to `hydramata-rels`. Then it doesn't need to be translated twice before being placed into the rels-ext slot in the accumulator.
